engine/runtime/src/main/scala/org/enso/compiler/pass/desugar/Imports.scala

```diff
@@ -0,0 +1,138 @@
+    * @param moduleContext a context object that contains the information needed
+    *                      to process a module
+    * @return `ir`, possibly having made transformations or annotations to that
+    *         IR.
+    */
+  override def runModule(
+    ir: IR.Module,
+    moduleContext: ModuleContext
+  ): IR.Module = {
+    val newImports = ir.imports.map {
+      case i: IR.Module.Scope.Import.Module =>
+        desugarCurrentProjectAlias(i.name, moduleContext)
+          .map { newName =>
+            val parts = newName.parts
+            if (parts.length == 2) {
+              i.copy(
+                name = newName.copy(parts = parts :+ mainModuleName),
+                rename = computeRename(
+                  i.rename,
+                  parts(1).asInstanceOf[IR.Name.Literal]
+                )
+              )
+            } else { i.copy(name = newName) }
+          }
+          .getOrElse(
+            IR.Error.ImportExport(
+              i,
+              IR.Error.ImportExport.ProjectKeywordUsedButNotInProject("import")
+            )
+          )
+      case other => other
+    }
+    val newExports = ir.exports.map {
+      case ex: IR.Module.Scope.Export.Module =>
+        desugarCurrentProjectAlias(ex.name, moduleContext)
+          .map { newName =>
+            val parts = newName.parts
+            if (parts.length == 2) {
+              ex.copy(
+                name = newName.copy(parts = parts :+ mainModuleName),
+                rename = computeRename(
+                  ex.rename,
+                  parts(1).asInstanceOf[IR.Name.Literal]
+                )
+              )
+            } else { ex.copy(name = newName) }
+          }
+          .getOrElse(
+            IR.Error.ImportExport(
+              ex,
+              IR.Error.ImportExport.ProjectKeywordUsedButNotInProject("export")
+            )
+          )
+      case other => other
+    }
+    ir.copy(imports = newImports, exports = newExports)
+  }
+
+  /** Executes the pass on the provided `ir`, and returns a possibly transformed
+    * or annotated version of `ir` in an inline context.
+    *
+    * @param ir            the Enso IR to process
+    * @param inlineContext a context object that contains the information needed
+    *                      for inline evaluation
+    * @return `ir`, possibly having made transformations or annotations to that
+    *         IR.
+    */
+  override def runExpression(
+    ir: IR.Expression,
+    inlineContext: InlineContext
+  ): IR.Expression = ir
+
+  private def computeRename(
+    originalRename: Option[IR.Name.Literal],
+    qualName: IR.Name.Literal
+  ): Some[IR.Name.Literal] = Some(originalRename.getOrElse(qualName))
+
+  val currentProjectAlias = "project"
+
+  private def desugarCurrentProjectAlias(
+    name: IR.Name.Qualified,
+    context: ModuleContext
+  ): Option[IR.Name.Qualified] = {
+    if (name.parts.length >= 0 && name.parts.head.name == currentProjectAlias) {
```

Comment:
This is a minor nitpick, but I'd suggest that this could be maybe cleaner?
```suggestion
    name.parts match {
        case head :: _ if head.name == currentProjectAlias =>
```